src/overlays/actors/ovl_En_Dns/z_en_dns.c

```diff
@@ -27,74 +52,603 @@ const ActorInit En_Dns_InitVars = {
+        Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    }
+    if ((this->unk_2C6 & 0x100) && (DECR(this->unk_2D0) == 0)) {
+        this->unk_2C6 &= ~0x100;
+        globalCtx->nextEntranceIndex = 0x5010;
+        gSaveContext.unk_3F4A = 0;
+        globalCtx->sceneLoadFlag = 0x14;
+        globalCtx->unk_1887F = 3;
+        gSaveContext.nextTransition = 3;
+    }
+}
+
+void func_8092D4D8(EnDns* this, GlobalContext* globalCtx) {
+    s16 sp2E = this->actor.yawTowardsPlayer;
+
+    if (ENDNS_PARAMS_4000(&this->actor) && (this->unk_2D2 == 0)) {
+        if (func_8092CE38(this)) {
+            func_8092C63C(this, 2);
+        }
+    } else if (func_8010BF58(&this->actor, globalCtx, this->unk_1E0, this->unk_2F4, &this->unk_1DC)) {
+        func_8013AED4(&this->unk_2C6, 3, 7);
+        this->unk_2F4 = NULL;
+        if (ENDNS_PARAMS_4000(&this->actor)) {
+            if (!(gSaveContext.eventInf[1] & 0x20)) {
+                this->skelAnime.animCurrentFrame = this->unk_2F0;
+                this->actor.world.rot.y = this->unk_2DA;
+                func_8092C63C(this, 8);
+            }
+            this->unk_2CC = 0;
+            this->unk_2CE = 0;
+            this->unk_2C6 |= 0x10;
+        }
+        this->actionFunc = func_8092D1B8;
+    } else {
+        Math_ApproachS(&this->actor.shape.rot.y, sp2E, 3, 0x2AA8);
+    }
+}
+
+void func_8092D5E8(EnDns* this, GlobalContext* globalCtx) {
+    static s32 D_8092DE0C[] = {
+        0,
+        0,
+        4,
+        6,
+    };
+
+    u32 temp_v0;
+    u32 temp_v1;
+
+    if (func_800EE29C(globalCtx, this->unk_2C8)) {
+        temp_v0 = func_800EE200(globalCtx, this->unk_2C8);
+        temp_v1 = globalCtx->csCtx.npcActions[temp_v0]->unk0;
+        if (this->unk_1D8 != (u8)temp_v1) {
+            func_8092C63C(this, D_8092DE0C[temp_v1]);
+            this->unk_1D8 = temp_v1;
+        }
+
+        if (((this->unk_2F8 == 4) || (this->unk_2F8 == 6)) &&
+            func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+            func_8092C63C(this, this->unk_2F8 + 1);
+        }
+
+        func_800EDF24(&this->actor, globalCtx, temp_v0);
+    }
+}
+
+void EnDns_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDns* this = THIS;
+
+    if (!func_8092D068(this)) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 18.0f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_06002DD8, NULL, this->jointTable, this->morphTable, 13);
+    this->unk_2F8 = -1;
+    func_8092C63C(this, 2);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(0x16), &sColChkInfoInit);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.targetMode = 0;
+    this->actor.gravity = -0.8f;
+    this->unk_2D2 = 0;
+    this->unk_2C6 = 0;
+    func_8013AED4(&this->unk_2C6, 3, 7);
+    this->unk_2C6 |= (0x40 | 0x10);
```

Comment:
This is an interesting one. Presumably this isn't required for matching?

Reply:
OR'ing the flags together? Engineer recommended me to do it in a previous PR.